lib/json-schema/util/uri.rb

```diff
@@ -1,7 +1,12 @@
+require 'singleton'
+
 module JSON
   module Util
-    module URI
-      def self.normalized_uri(uri)
+    class URI
+      include Singleton
+      attr_accessor :normalize_cache, :parse_cache
```

Comment:
Also, it might be good to make these accessors "protected" methods (they're globally available caches, so it would be good to prevent access from outside this class)